tests/processors/data_contributions/test_processor.py

```diff
@@ -0,0 +1,125 @@
+        processor.process({}, conn)
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        document = conn['database']['documents'].find_one(id=updated_contrib['document_id'])
+
+        updated_contrib.update({'approved': False})
+        conn['explorer']['data_contributions'].update(updated_contrib, ['id'])
+        processor.process({}, conn)
+
+        assert conn['database']['documents'].find_one(id=document['id']) is None
+
+
+    def test_updates_contribution_document_id_if_document_is_removed(self, conn, valid_data_contrib):
+        processor.process({}, conn)
+        processed_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+
+        processed_contrib.update({'approved': False})
+        conn['explorer']['data_contributions'].update(processed_contrib, ['id'])
+        processor.process({}, conn)
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+
+        assert processed_contrib['document_id'] is not None
+        assert updated_contrib['document_id'] is None
+
+
+    def test_ignores_contribution_with_invalid_url(self, conn, valid_data_contrib):
+        contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        contrib_attrs = {
+            'url': 'invalid url',
+        }
+        contrib.update(contrib_attrs)
+        conn['explorer']['data_contributions'].update(contrib, ['id'])
+        processor.process({}, conn)
+
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        assert updated_contrib['document_id'] is None
+
+
+    def test_ignores_archive_contribution(self, conn, valid_data_contrib):
+        contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        contrib_attrs = {
+            'url': 'https://github.com/opentrials/processors/archive/master.zip',
+        }
+        contrib.update(contrib_attrs)
+        conn['explorer']['data_contributions'].update(contrib, ['id'])
+        processor.process({}, conn)
+
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        assert updated_contrib['document_id'] is None
+
+
+    def test_ignores_contribution_with_invalid_trial_id(self, conn, valid_data_contrib):
+        contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        contrib_attrs = {
+            'trial_id': 'e0f3ef3c-edba-11e6-ae32-e4b3181a2c8c',
+        }
+        contrib.update(contrib_attrs)
+        conn['explorer']['data_contributions'].update(contrib, ['id'])
+        processor.process({}, conn)
+
+        updated_contrib = conn['explorer']['data_contributions'].find_one(id=valid_data_contrib)
+        assert updated_contrib['document_id'] is None
+
+
+@pytest.fixture
+def valid_data_contrib(conn, data_contribution, document_category, trial):
```

Comment:
Yes, a valid data contribution passes [this query](https://github.com/opentrials/processors/pull/119/commits/707b0f5207343e37bda261b5882058e0f3ba2c6e#diff-afa5df5f8a3a258e5bfa03a15f836566R29).